Hmmm, the test failure in a single configuration is real. Which is bizarre.

https://github.com/napari/napari/actions/runs/10627301445/job/29460558796?pr=7228#step:13:473

```
  _________________________________ test_polygon _________________________________
  
      def test_polygon():
          """Test creating Shape with a random polygon."""
          # Test a single non convex six vertex polygon
          data = np.array(
              [
                  [10.97627008, 14.30378733],
                  [12.05526752, 10.89766366],
                  [8.47309599, 12.91788226],
                  [8.75174423, 17.83546002],
                  [19.27325521, 7.66883038],
                  [15.83450076, 10.5778984],
              ]
          )
          shape = Polygon(data)
          np.testing.assert_array_equal(shape.data, data)
          assert shape.data_displayed.shape == (6, 2)
          assert shape.slice_key.shape == (2, 0)
          # should get few triangles
          expected_face = (8, 2)
          assert shape._edge_vertices.shape == (16, 2)
  >       assert shape._face_vertices.shape == expected_face
  E       assert (6, 2) == (8, 2)
  E         
  E         At index 0 diff: 6 != 8
  E         
  E         Full diff:
  E           (
  E         -     8,
  E         ?     ^
  E         +     6,
  E         ?     ^
  E               2,
  E           )
  
  napari/layers/shapes/_shapes_models/_tests/test_shapes_models.py:101: AssertionError
```

--- a/napari/layers/shapes/_shapes_models/_tests/test_shapes_models.py
+++ b/napari/layers/shapes/_shapes_models/_tests/test_shapes_models.py
@@ -1,3 +1,5 @@
+import sys
+
 import numpy as np
 import pytest
 from vispy.geometry import PolygonData
@@ -96,7 +98,7 @@ def test_polygon():
     assert shape.data_displayed.shape == (6, 2)
     assert shape.slice_key.shape == (2, 0)
     # should get few triangles
-    expected_face = (8, 2)
+    expected_face = (6, 2) if 'triangle' in sys.modules else (8, 2)
     assert shape._edge_vertices.shape == (16, 2)
     assert shape._face_vertices.shape == expected_face
 

